operator/internal/manifests/env_vars.go

```diff
@@ -0,0 +1,111 @@
+package manifests
+
+import (
+	"strings"
+
+	"github.com/imdario/mergo"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	httpProxyKey  = "HTTP_PROXY"
+	httpsProxyKey = "HTTPS_PROXY"
+	noProxyKey    = "NO_PROXY"
+)
+
+var ProxyEnvNames = []string{
+	httpProxyKey,
+	strings.ToLower(httpProxyKey),
+	httpsProxyKey,
+	strings.ToLower(httpsProxyKey),
+	noProxyKey,
+	strings.ToLower(noProxyKey),
+}
+
+func configureProxyEnv(pod *corev1.PodSpec, opts Options) error {
+	for _, envVar := range ProxyEnvNames {
+		resetProxyVar(pod, envVar)
+	}
+
+	proxySpec := opts.Stack.Proxy
+	if proxySpec == nil {
+		return nil
+	}
+
+	src := corev1.Container{
+		Env: ToEnvVars(proxySpec.HTTPProxy, proxySpec.HTTPSProxy, proxySpec.NoProxy),
+	}
+
+	for i, dst := range pod.Containers {
+		if err := mergo.Merge(&dst, src, mergo.WithAppendSlice); err != nil {
+			return err
+		}
+		pod.Containers[i] = dst
+	}
+
+	return nil
+}
+
+func resetProxyVar(podSpec *corev1.PodSpec, name string) {
+	for i, container := range podSpec.Containers {
+		found, index := getEnvVar(name, container.Env)
+		if found {
+			podSpec.Containers[i].Env = append(podSpec.Containers[i].Env[:index], podSpec.Containers[i].Env[index+1:]...)
+		}
+	}
+}
+
+// getEnvVar matches the given name with the envvar name
+func getEnvVar(name string, envVars []corev1.EnvVar) (bool, int) {
+	for i, env := range envVars {
+		if env.Name == name || env.Name == strings.ToLower(name) {
```

Comment:
The `ToLower` is unnecessary, because the list of variables already includes both permutations.